From b3316126fa5fc371f3880b167e1a8d409c9a87f7 Mon Sep 17 00:00:00 2001
From: <temporary>
Date: Tue, 11 Jul 2023 12:00:00
Subject: [PATCH] Patch for bootstrapping a helperlib.ll for use in PIXIE.

As title.
---
 numba/_helperlib.c |  6 ------
 numba/_helpermod.c | 52 ----------------------------------------------------
 2 files changed, 58 deletions(-)

diff --git a/numba/_helperlib.c b/numba/_helperlib.c
--- a/numba/_helperlib.c
+++ b/numba/_helperlib.c
@@ -1251,9 +1251,3 @@ numba_gdb_breakpoint(void) {
  */
 
 #include "_lapack.c"
-
-/*
- * PRNG support
- */
-
-#include "_random.c"
diff --git a/numba/_helpermod.c b/numba/_helpermod.c
--- a/numba/_helpermod.c
+++ b/numba/_helpermod.c
@@ -115,14 +115,6 @@ build_c_helpers_dict(void)
     declmethod(xgesv);
     declmethod(xxnrm2);
 
-    /* PRNG support */
-    declmethod(get_py_random_state);
-    declmethod(get_np_random_state);
-    declmethod(get_internal_random_state);
-    declmethod(rnd_shuffle);
-    declmethod(rnd_init);
-    declmethod(poisson_ptrs);
-
     /* Unicode string support */
     declmethod(extract_unicode);
     declmethod(gettyperecord);
@@ -131,42 +123,6 @@ build_c_helpers_dict(void)
     /* for gdb breakpoint */
     declmethod(gdb_breakpoint);
 
-    /* for dictionary support */
-    declmethod(test_dict);
-    declmethod(dict_new_sized);
-    declmethod(dict_set_method_table);
-    declmethod(dict_free);
-    declmethod(dict_length);
-    declmethod(dict_lookup);
-    declmethod(dict_insert);
-    declmethod(dict_insert_ez);
-    declmethod(dict_delitem);
-    declmethod(dict_popitem);
-    declmethod(dict_iter_sizeof);
-    declmethod(dict_iter);
-    declmethod(dict_iter_next);
-    declmethod(dict_dump);
-
-    /* for list support */
-    declmethod(test_list);
-    declmethod(list_new);
-    declmethod(list_set_method_table);
-    declmethod(list_free);
-    declmethod(list_base_ptr);
-    declmethod(list_size_address);
-    declmethod(list_length);
-    declmethod(list_allocated);
-    declmethod(list_is_mutable);
-    declmethod(list_set_is_mutable);
-    declmethod(list_setitem);
-    declmethod(list_getitem);
-    declmethod(list_append);
-    declmethod(list_delitem);
-    declmethod(list_delete_slice);
-    declmethod(list_iter_sizeof);
-    declmethod(list_iter);
-    declmethod(list_iter_next);
-
 #define MATH_UNARY(F, R, A) declmethod(F);
 #define MATH_BINARY(F, R, A, B) declmethod(F);
     #include "mathnames.h"
@@ -193,13 +149,6 @@ _numba_flush_stdout(void) {
 
 
 static PyMethodDef ext_methods[] = {
-    { "rnd_get_state", (PyCFunction) _numba_rnd_get_state, METH_O, NULL },
-    { "rnd_get_py_state_ptr", (PyCFunction) _numba_rnd_get_py_state_ptr, METH_NOARGS, NULL },
-    { "rnd_get_np_state_ptr", (PyCFunction) _numba_rnd_get_np_state_ptr, METH_NOARGS, NULL },
-    { "rnd_seed", (PyCFunction) _numba_rnd_seed, METH_VARARGS, NULL },
-    { "rnd_set_state", (PyCFunction) _numba_rnd_set_state, METH_VARARGS, NULL },
-    { "rnd_shuffle", (PyCFunction) _numba_rnd_shuffle, METH_O, NULL },
-    { "_import_cython_function", (PyCFunction) _numba_import_cython_function, METH_VARARGS, NULL },
     { NULL },
 };
 
@@ -271,7 +220,6 @@ MOD_INIT(_helperlib) {
     PyModule_AddIntConstant(m, "py_unicode_2byte_kind", PyUnicode_2BYTE_KIND);
     PyModule_AddIntConstant(m, "py_unicode_4byte_kind", PyUnicode_4BYTE_KIND);
     PyModule_AddIntConstant(m, "py_unicode_wchar_kind", PyUnicode_WCHAR_KIND);
-    numba_rnd_ensure_global_init();
 
     return MOD_SUCCESS_VAL(m);
 }
